controllers/dopplersecret_controller_secrets.go

```diff
@@ -276,6 +289,18 @@ func (r *DopplerSecretReconciler) UpdateSecret(ctx context.Context, dopplerSecre
 		requestedSecretVersion = ""
 	}
 
+	if existingKubeSecret != nil {
+		// Compare existing managed secret labels to labels defined in the doppler secret.
+		// If they are not equal, we will update the managed secret with the new labels.
+		if reflect.DeepEqual(existingKubeSecret.Labels, GetKubeSecretLabels(dopplerSecret.Spec.ManagedSecretRef.Labels)) {
+			log.Info("[-] Managed secret labels not modified.")
+		} else {
+			// If labels have changed, set requestedSecretVersion to an empty secret version to reload the secrets.
+			log.Info("[/] Labels changed, reloading secrets.")
+			requestedSecretVersion = ""
```

Comment:
I think this might be a good opportunity to refactor the `UpdateSecret` logic. Let me know what you think of the attached patch.